### PR TITLE
external/stdk: Porting st-device-sdk on TizenRT

### PR DESCRIPTION
**Pull request by SRCN-LiQin**
1. Add st-device-sdk in external
2. Add smart lamp example base on st-device-sdk

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [d85b42b19850b77173c923566d448f8faf45b89a] - Code Rule Check Result:
apps/examples/stdk_smart_lamp/device_control.c:176: ERROR: [BRC_M_SMT] that open brace { should be on the previous line
apps/examples/stdk_smart_lamp/smart_lamp.c:28: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
apps/examples/stdk_smart_lamp/smart_lamp.c:28: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
apps/examples/stdk_smart_lamp/smart_lamp.c:29: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
apps/examples/stdk_smart_lamp/smart_lamp.c:29: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
apps/examples/stdk_smart_lamp/smart_lamp.c:29: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
apps/examples/stdk_smart_lamp/smart_lamp.c:107: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
apps/examples/stdk_smart_lamp/smart_lamp.c:123: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
apps/examples/stdk_smart_lamp/smart_lamp.c:143: ERROR: [IDT_M_SCH] switch and case should be at the same indent
apps/examples/stdk_smart_lamp/smart_lamp.c:143: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
apps/examples/stdk_smart_lamp/smart_lamp.c:179: ERROR: [BRC_M_SMT] that open brace { should be on the previous line
apps/examples/stdk_smart_lamp/smart_lamp.c:179: ERROR: [SPC_M_KWD] space required before the open parenthesis '('

**Comment by seinfra** [comment]
Target : [d85b42b19850b77173c923566d448f8faf45b89a] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [0236e63b80b3771854327c78c10705b94c4ae908] - Code Rule Check Result:
apps/examples/stdk_smart_lamp/smart_lamp.c:29: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)

**Comment by seinfra** [comment]
Target : [0236e63b80b3771854327c78c10705b94c4ae908] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [393b0ee7bdd6d84b6e90f5f7e2d59baf82201013] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [393b0ee7bdd6d84b6e90f5f7e2d59baf82201013] - Code Rule Check OK.

File: external/Kconfig [hunk 1]
```diff
@@ -27,3 +27,6 @@ source "$EXTERNALDIR/dhcpc/Kconfig.protocol"
 source "$EXTERNALDIR/vec/Kconfig"
 source "$EXTERNALDIR/lzma/Kconfig"
 source "$EXTERNALDIR/nanopb/Kconfig"
+source "$EXTERNALDIR/libsodium/Kconfig"
+source "$EXTERNALDIR/openssl/Kconfig"
```
Comment (zhouxinhe) [inline external/Kconfig]: How about move these two lines to PR #4000 and #4001 ?

**Comment by seinfra** [comment]
Target : [733127cf97bd424eca2a7ab16711b338c6a188ac] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [0aa3e5b5b3bf390936cc18b545cfab2060ad3f37] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [0aa3e5b5b3bf390936cc18b545cfab2060ad3f37] - Code Rule Check OK.

**Pull request merged**